packages/validator/src/services/indices.ts

```diff
@@ -83,10 +83,8 @@ export class IndicesService {
       return this.pollValidatorIndicesPromise;
     }
 
-    this.pollValidatorIndicesPromise = this.pollValidatorIndicesInternal(pubkeysHex);
-    // Once the pollValidatorIndicesInternal() resolves or rejects null the cached promise so it can be called again.
-    // eslint-disable-next-line @typescript-eslint/no-floating-promises
-    this.pollValidatorIndicesPromise.finally(() => {
+    this.pollValidatorIndicesPromise = this.pollValidatorIndicesInternal(pubkeysHex).finally(() => {
```

Comment:
this will not save one from uncaught exception see this example which is what models rejection here:
![image](https://github.com/ChainSafe/lodestar/assets/76567250/52b9d69c-bde3-43cc-889f-6d3b2a3d1fbf)

Reply:
2) you have eaten up the log if your finally would have "caught" the rejection

Reply:
the error has to be handled by the caller as we do it for all function calls and it is handled by all callers

Reply:
the issue was that the finally was attached separately to the promise, and it's resolved in a different context. See the eslint warning about floating promises

Reply:
The main issue with the inner-catch handler is that `pollValidatorIndices` pretends to have worked find and just not discovered any new indicies which is incorrect, as upstream code relies on this to either work properly or throw. e.g. example in doppelganger service

Reply:
> 2. you have eaten up the log if your finally would have "caught" the rejection

which log? `finally` does not catch exceptions

Reply:
put this in a file and run node filename: (by catch i meant unhandled exception )

```
async function rejectMe(){
        let myPromise = Promise.reject(1).finally(()=>{
                myPromise = null })
}

async function callRejectMe(){
        try{
                await rejectMe().catch(e=>{console.log("caught error1")})
        }catch(e){
                console.log("caught 2")
        }
}

setTimeout(()=>{callRejectMe()})
setTimeout(()=>{console.log("finished")},100000)
```

Reply:
> The main issue with the inner-catch handler is that `pollValidatorIndices` pretends to have worked fine and just not discovered any new indicies which is incorrect, as upstream code relies on this to either work properly or throw. e.g. example in doppelganger service

i don't think what has it to do with doppleganger? if you don't have indices for some pubkeys, you you will not mark them correct for doppleganger.

if pollIndices failed for the validators for which you have no indices, then you (one) should just proceed with other validators for which you have indices.

Reply:
doppleganger code:
```ts
const newIndices = await this.indicesService.pollValidatorIndices(pubkeysToCheckWithoutIndex);
    for (const index of newIndices) {
      const pubkey = this.indicesService.index2pubkey.get(index);
      if (pubkey) {
        indicesToCheckMap.set(index, pubkey);
      }
    }
    ```
    
    so it this function doesnt return new indices, it doesn't even starts its monitoring, so how will a pubkey even vote?

Reply:
```ts
async function rejectMe(){
        let myPromise = Promise.reject(1).finally(()=>{
                myPromise = null })
}
```

doesn't even return the promise, it's also not clear to me what you wanna show case here. We have a lot of places in the code that throw error but we decide to handle it further upstream.

e.g. why not handle the error directly when calling `getStateValidators`
https://github.com/ChainSafe/lodestar/blob/7e35c924e7cf3ccc119f80f81f6de40c70ff4d29/packages/validator/src/services/indices.ts#L129

we could also set it to an empty array right there..but again why change the behavior, if we fail to fetch validators it should error, and let caller decide what do to with it.

Reply:
> doppleganger code:

right, seems like we would keep the pubkey idle for one more epoch as we don't check it. Still doesn't change my point, although makes it less critical that we don't pretend fetching worked in case of an error

Reply:
> But this is a dangerous code pattern which must be avoided

you have that literally we every promise we create, that's why we have an eslint rule for floating promises